File: arb-www/src/schema/rollingInsights/nba/schedule.ts
import { BaseDTO } from '../../BaseDTO';

/**
 * NBA Rolling Insights Schedule Game
 */
export class NBARollingInsightsScheduleGame extends BaseDTO {
  away_team!: string;
  home_team!: string;
  'away_team_ID'!: number;
  'home_team_ID'!: number;
  'game_ID'!: string;
  'game_time'!: string;
  'season_type'!: string;
  'event_name'?: string | null;
  round?: string | null;
  season!: string;
  status!: string;
  broadcast?: string | null;
}

/**
 * Rolling Insights NBA Schedule response wrapper
 */
export interface NBARollingInsightsScheduleResponse {
  league: string;
  data_type: string;
  data: {
    Nba: {
      Schedule: NBARollingInsightsScheduleGame[];
    };
  };
  filtered_count: number;
  total_count: number;
}

File: arb-rs/src/schema/rolling_insights/nba/schedule.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NBARollingInsightsScheduleGame {
    pub away_team: String,
    pub home_team: String,
    #[serde(rename = "away_team_ID")]
    pub away_team_id: i32,
    #[serde(rename = "home_team_ID")]
    pub home_team_id: i32,
    #[serde(rename = "game_ID")]
    pub game_id: String,
    #[serde(rename = "game_time")]
    pub game_time: String,
    #[serde(rename = "season_type")]
    pub season_type: String,
    #[serde(rename = "event_name")]
    pub event_name: Option<String>,
    pub round: Option<String>,
    pub season: String,
    pub status: String,
    pub broadcast: Option<String>,
}
